### ISSUE TITLE
Cannot save settings: Please specify a different SDK name

### ISSUE THREAD

**Issue by sugresmax** [KronicDeth/intellij-elixir#1522]
I,m added Elixir plugin to my RubyMine, but i can't configure it. Always displayed error on apply settings: 
> Cannot save settings: Please specity a different SDK name.

```~$ elixir -v
Erlang/OTP 22 [erts-10.4.4] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe]

Elixir 1.9.0 (compiled with Erlang/OTP 20)
```
```
RubyMine 2019.1.3
Build #RM-191.7479.66, built on June 27, 2019
Licensed to ***
Subscription is active until September 9, 2019
JRE: 1.8.0_202-release-1483-b53 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.9.0-9-amd64
```

**Comment by KronicDeth** [comment]
For some package managers the SDK name is inferred from the install path.  What is the install path of your Erlang and Elixir?

**Comment by sugresmax** [comment]
`/home/sugresmax/.asdf/installs/elixir/1.9.0`
`/home/sugresmax/.asdf/installs/erlang/22.0.7`

**Comment by qw3r** [comment]
Hi @sugresmax, have you managed to find the problem (or a workaround)?

**Comment by sugresmax** [comment]
Hi @qw3r 
No

**Comment by sebastian-schuth** [comment]
@qw3r @KronicDeth 

I am running into this, too.

IDE: RubyMine 2019.3.4
Elixir Plugin Version: 11.6.0

I manage elixir and erlang installs using `asdf`, which puts elixir into `~/.asdf/installs/elixir/{VERSION}` and erlang into `~/.asdf/installs/erlang/{VERSION}`.

Following locations are shown in the paths overview ("Classpath") in IntelliJ or, in my case, RubyMine. See the attached screenshot for more context how this shows up.

```
file:///Users/sschuth/.asdf/installs/elixir/1.10/lib/logger/ebin
file:///Users/sschuth/.asdf/installs/elixir/1.10/lib/elixir/ebin
file:///Users/sschuth/.asdf/installs/elixir/1.10/lib/ex_unit/ebin
file:///Users/sschuth/.asdf/installs/elixir/1.10/lib/mix/ebin
file:///Users/sschuth/.asdf/installs/elixir/1.10/lib/eex/ebin
file:///Users/sschuth/.asdf/installs/elixir/1.10/lib/iex/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/erl_interface-3.13.2/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/eunit-2.4.1/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/diameter-2.2.3/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/tftp-1.0.2/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/public_key-1.7.2/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/parsetools-2.1.8/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/inets-7.1.2/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/kernel-6.5.2/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/megaco-3.18.8/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/hipe-3.19.3/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/runtime_tools-1.14/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/crypto-4.6.5/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/erl_docgen-0.11/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/syntax_tools-2.2.1/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/asn1-5.0.10/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/odbc-2.12.4/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/debugger-4.2.8/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/sasl-3.4.2/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/wx-1.9/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/os_mon-2.5.1/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/ftp-1.0.4/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/erts-10.7/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/dialyzer-4.1.1/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/ssl-9.6/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/jinterface-1.10.1/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/tools-3.3.1/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/snmp-5.5/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/et-1.6.4/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/ssh-4.9/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/reltool-0.8/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/compiler-7.5.3/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/mnesia-4.16.3/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/eldap-1.2.8/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/common_test-1.18.2/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/stdlib-3.12/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/observer-2.9.3/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/edoc-0.11/ebin
file:///Users/sschuth/.asdf/installs/erlang/22.3/lib/xmerl-1.3.23/ebin
```

As you can see in the screenshot, the SDKs are reported to have the same name as the already installed elixir/erlang SDKs, but they show up under different names (at least that is how I interpret what I am seeing in the UI).

To make things a bit more clear, these are the paths used for the "old" SDK, pointing to a `brew`-installed version of elixir and erlang:

```
file:///usr/local/Cellar/elixir/1.8.1/lib/logger/ebin
file:///usr/local/Cellar/elixir/1.8.1/lib/elixir/ebin
file:///usr/local/Cellar/elixir/1.8.1/lib/ex_unit/ebin
file:///usr/local/Cellar/elixir/1.8.1/lib/mix/ebin
file:///usr/local/Cellar/elixir/1.8.1/lib/eex/ebin
file:///usr/local/Cellar/elixir/1.8.1/lib/iex/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/mnesia-4.15.6/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/sasl-3.3/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/parsetools-2.1.8/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/syntax_tools-2.1.7/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/erl_docgen-0.9/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/reltool-0.7.8/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/snmp-5.2.12/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/compiler-7.3.2/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/eunit-2.3.7/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/os_mon-2.4.7/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/kernel-6.3.1/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/tools-3.1/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/inets-7.0.7/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/debugger-4.2.6/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/xmerl-1.3.20/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/ssl-9.2.2/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/observer-2.9/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/wx-1.8.7/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/diameter-2.2.1/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/dialyzer-3.3.2/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/ftp-1.0.2/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/erl_interface-3.11.2/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/hipe-3.18.3/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/tftp-1.0.1/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/common_test-1.17.1/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/erts-10.3.4/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/et-1.6.4/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/stdlib-3.8.1/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/eldap-1.2.6/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/megaco-3.18.4/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/runtime_tools-1.13.2/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/edoc-0.10/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/ssh-4.7.6/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/public_key-1.6.5/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/asn1-5.0.8/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/crypto-4.4.2/ebin
file:///usr/local/Cellar/erlang/21.3.7/lib/erlang/lib/otp_mibs-1.2.1/ebin
```

Added SDK:
![grafik](https://user-images.githubusercontent.com/43844754/80910865-c323b580-8d32-11ea-8047-a936e78cb9c6.png)

Pre-existing (and working) SDK:
![grafik](https://user-images.githubusercontent.com/43844754/80910888-e484a180-8d32-11ea-9171-916f450d1ed1.png)

Let me know if I can help you in any way with fixing/debugging this.

**Comment by sebastian-schuth** [comment]
Strangely enough, it looks like one can work around this by pressing [Apply] and then [Cancel] -- I see both SDKs listed, but pressing [OK] does not work.

**Comment by the-kid89** [comment]
Why is this broken though?

**Comment by sebastian-schuth** [comment]
> Why is this broken though?

I do not get what you want to say -- can you rephrase that question?

**Comment by mattgibson** [comment]
Just ran into this and although it will save by using apply and cancel, this also stops all other unrelated changes to any of the Rubymine settings. I have to use apply and cancel after any change anywhere. The 'OK' button takes me to the erlang setting page and displays the "Please specify a different SDK name" message.

**Comment by mattgibson** [comment]
It seems the SDKs are missing after the IDE is restarted, so the apply and cancel trick is no good. 

The install locations for me are:

`/Users/matthewgibson/.asdf/installs/elixir/1.11.2`
`/Users/matthewgibson/.asdf/installs/erlang/23.1.4`

Using Rubymine 2020.3

@KronicDeth Is this issue still requiring more feedback? I'm not sure what other details I can supply. Happy to help debug it if necessary.

**Comment by sebastian-schuth** [comment]
Note: This happens with brew-installed Elixir, too, not only with `asdf`-managed versions.

I am on RubyMine 2020.3.2 in the meantime.

**Comment by sebastian-schuth** [comment]
It also complaints about the SDK being corrupted -- it looks like something fails to automatically determine the name of the SDK.

**Comment by the-kid89** [comment]
Is there a work around to get Elixir to work when you run into this problem yet?

**Comment by Slawikus** [comment]
Facing the same problem with RubyMine 2022.1.1, Elixir 1.13.4-otp-24, Erlang 24.3.4, plugin version 13.0.0

**Comment by KronicDeth** [comment]
How the SDK is saved in Small IDEs (like Rubymine) has been changed in a #2681.  Closing this in favor of a re-open if it still occurs with that change.

@the-kid89 the work-around for any bug that only affects the Small IDEs like Rubymine is to use IntelliJ Community or IntelliJ Ultimate where the SDK can be stored directly in an Elixir module
instead of on the side in a facet of a Ruby module
